components/events/Add.vue

```diff
@@ -1,91 +1,58 @@
 <script setup lang="ts">
-import { storeToRefs } from "pinia";
-import { getAllEvents, EventDifficulty } from "./utils";
+import { EventDifficulty } from "./utils";
 import type { EventType } from "./utils";
-import { useAuthStore } from "~/stores/auth";
 
-const showModal = ref(false);
-
-const { jwt, isLoggedIn } = storeToRefs(useAuthStore());
-
-const inputStartTime = ref("");
-const inputEndTime = ref("");
-
-const formData = ref<Omit<EventType, "id">>({
-  name: "",
-  location: "",
-  mazemapLink: "",
-  summary: "",
-  description: "",
-  slides: "",
-  image: "",
-  organizer: "",
-  unixStartTime: 0,
-  unixEndTime: 0,
-  difficulty: EventDifficulty.EASY, // Add the difficulty field to formData
+const props = defineProps<
+  | {
+      isEdit: false;
+      formValues?: formWithoutId;
+    }
+  | { formValues: EventType; isEdit: true }
+>();
+
+const emit = defineEmits<{
+  eventAdded: [values: formWithoutId];
+  eventEdited: [values: EventType];
+}>();
+
+type formType = Omit<EventType, "id"> & { id?: string };
+type formWithoutId = Omit<EventType, "id">;
```

Comment:
Can we move these type declarations up in the file to make it clearer where `formWithoutId` is coming from for the props and emits?